src/CBot/CBotInstr/CBotDefArray.cpp

```diff
@@ -96,17 +103,23 @@ CBotInstr* CBotDefArray::Compile(CBotToken* &p, CBotCStack* pStack, CBotTypResul
 
         if (IsOfType(p, ID_ASS))                                        // with an assignment
         {
-            if ((inst->m_listass = CBotTwoOpExpr::Compile(p, pStk)) != nullptr)
+            pStk->SetStartError(p->GetStart());
+            if ( IsOfType(p, ID_SEP) )
             {
-                if (!pStk->GetTypResult().Compare(type))  // compatible type ?
-                {
-                    pStk->SetError(CBotErrBadType1, p->GetStart());
-                    goto error;
-                }
+                pStk->SetError(CBotErrBadLeft, p->GetPrev());
```

Comment:
This should be a new error type, "expression expected after =" (in all the CBotDef* classes)